### PR TITLE
make parseCookies parse cookies as rawCookies

### LINKED ISSUE #1228: another cookie parsing problems
```
dub init -t vibe.d test
cd test
dub run &
curl localhost:8080 -H 'Cookie: __context={"SLT":"%f,*]","UID":""};'
```
and watch it crash:

```
400 - Bad Request

Bad Request

Internal error information:
object.Exception@../../../.dub/packages/vibe-d-0.7.24/source/vibe/textfilter/urlencode.d(179): invalid percent encoding
----------------
5   cookiefuckup                        0x000000010ca3e9e3 pure @safe bool std.exception.enforce!(Exception, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) + 107
6   cookiefuckup                        0x000000010cb96b09 pure @safe void vibe.textfilter.urlencode.filterURLDecode!(std.array.Appender!(immutable(char)[]).Appender).filterURLDecode(ref std.array.Appender!(immutable(char)[]).Appender, immutable(char)[], bool) + 257
7   cookiefuckup                        0x000000010cb96280 @safe immutable(char)[] vibe.textfilter.urlencode.urlDecode(immutable(char)[]) + 112
8   cookiefuckup                        0x000000010cb7903d void vibe.http.server.parseCookies(immutable(char)[], ref vibe.http.common.CookieValueMap) + 369
9   cookiefuckup                        0x000000010cb7766b bool vibe.http.server.handleRequest(vibe.core.stream.Stream, vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo, ref vibe.http.server.HTTPServerSettings, ref bool) + 4527
10  cookiefuckup                        0x000000010cb763bd void vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo) + 425
11  cookiefuckup                        0x000000010cb75da6 void vibe.http.server.listenHTTPPlain(vibe.http.server.HTTPServerSettings).doListen(vibe.http.server.HTTPListenInfo, bool).__lambda3(vibe.core.net.TCPConnection) + 54
12  cookiefuckup                        0x000000010cada05b void vibe.core.drivers.libevent2_tcp.ClientTask.execute() + 1191
13  cookiefuckup                        0x000000010cacd644 void vibe.core.core.makeTaskFuncInfo!(void delegate()).makeTaskFuncInfo(ref void delegate()).callDelegate(vibe.core.core.TaskFuncInfo*) + 120
14  cookiefuckup                        0x000000010caca9e7 void vibe.core.core.CoreTask.run() + 299
15  cookiefuckup                        0x000000010cbbb3f0 void core.thread.Fiber.run() + 44
16  cookiefuckup                        0x000000010cbbafd6 fiber_entryPoint + 98
17  ???                                 0x0000000000000000 0x0 + 0

```

not sure if simply urldecoding  is the  correct thing to do. there is a few rfc, notably i found
http://tools.ietf.org/html/rfc6265 via http://stackoverflow.com/questions/1969232/allowed-characters-in-cookies

### PR DESCRIPTION
**Pull request by yannick**
replaces #1336 
fixes #1228

### CODE REVIEW

File: source/vibe/http/server.d [hunk 1]
```diff
@@ -1863,17 +1863,36 @@ private void parseRequestHeader(HTTPServerRequest req, InputStream http_stream,
...
-
-		for (idx = 0; idx < str.length && str[idx] != ';'; idx++) {}
-		string value = str[0 .. idx].strip();
-		cookies[name] = urlDecode(value);
-		str = idx < str.length ? str[idx+1 .. $] : null;
-	}
+	import std.encoding : sanitize;
+	import std.array : split;
+	import std.string : strip;
+	import std.algorithm.iteration : map, filter, each;
```
Comment (s-ludwig) [inline source/vibe/http/server.d]: Should still be just `import std.algorithm : ...` to make DMD 2.066.x happy.

Comment (s-ludwig) [inline source/vibe/http/server.d]: Scratch that! I'll remove 2.066.0 support instead now that the release is out.

**Comment by s-ludwig** [comment]
Okay, should pass now. Merging.

**Comment by yannick** [comment]
:dancers: thanks

**Pull request merged**